### PR TITLE
Migrate to CircleCI 2.0

### PR DESCRIPTION
**Pull request by autrilla**
Build works on my fork (https://circleci.com/gh/autrilla/server-syncstorage/4)

### CODE REVIEW

**Comment by autrilla** [comment]
The docker version could be abstracted with an env var, but i don't think that's doable for the ubuntu image version, if that's what you mean

**Comment by autrilla** [comment]
I've parametrized the docker version, `DOCKER_VERSION` will have to be set to `17.12.1-ce` on CircleCI

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            "$CIRCLE_PROJECT_REPONAME" \
+            "$CIRCLE_BUILD_URL" > version.json
+      - store_artifacts:
+          path: version.json
+      # FIXME: should use an image w/ docker installed by default
+      - run:
+          name: Install Docker
+          command: |
+            set -x
+            curl -L -o /tmp/docker-$DOCKER_VERSION.tgz https://download.docker.com/linux/static/stable/x86_64/docker-$DOCKER_VERSION.tgz
```
Comment (rfk) [inline .circleci/config.yml]: Is there any way we can integrity-check the result of this download, even if it's just comparing its `sha256sum` against a known good value?

I guess the app itself doesn't integrity-check its dependencies either, but it probably should...

Comment (autrilla) [reply .circleci/config.yml]: I've added a check for the sha256 of the download

**Comment by autrilla** [comment]
@rfk can this be merged?

**Comment by rfk** [comment]
👍

**Pull request merged**